data-model/src/main/java/io/micronaut/data/model/naming/NamingStrategy.java

```diff
@@ -97,9 +98,11 @@ public interface NamingStrategy {
         if (property instanceof Association) {
             return mappedName((Association) property);
         } else {
-            return property.getAnnotationMetadata()
+            AnnotationMetadata propertyAnnotationMetadata = property.getAnnotationMetadata();
```

Comment:
This changes should be in Cosmos SQL builder

Reply:
I see no simple way and doing this will lead to duplicate logic and maintenance in NamingStrategy and CosmosSqlQueryBuilder as this logic cannot be easily decoupled. At least I don't see how, since we need to deal with 3 methods and check for property and association mapping.

Reply:
Maybe not setting this value at all https://github.com/micronaut-projects/micronaut-data/blob/master/data-processor/src/main/java/io/micronaut/data/processor/visitors/MappedEntityVisitor.java#L252 and then we would not have to deal with it

Reply:
I have added change here https://github.com/micronaut-projects/micronaut-data/pull/1760/commits/a7668c1625a02538838810170d6beba9888149b7 but still need to keep logic for getting generated mapped name in NamingStrategy, only added option to tell from the builder whether to use generated (default yes) or not (from cosmos queries)

Reply:
Why do you need to keep it? That method is mostly a helper working with the naming strategy, it should possible to extract it.

Reply:
Well there are 3 methods in NamingStrategy that need to change and not sure if possible to extract them because it is being called from various places. If I understand what you are suggesting. This current change looks safe to me, keeps old behavior and adds ability to override behavior for sql builder(s).

Reply:
Moved changes about "generated" attribute from NamingStrategy to CosmosSqlQueryBuilder.